lessson-01-assignment.txt

```diff
@@ -0,0 +1,179 @@
+  Student Load (0.3ms)  SELECT "students".* FROM "students" WHERE "students"."id" = 43 LIMIT 1 /*application='DocRails'*/
+=> 
+#<Student:0x000000010d8158e0
+...
+doc-rails(dev)> student.email
+=> "yvonne@kiehn.example"
+doc-rails(dev)> 
+
+Task 2 commands:
+25, trenton_haley@okuneva-tremblay.test
+27, vincenzo.marks@watsica.example
+
+Step1: get enrollment_id which course_id is 5 and final_grade is nil from Enrollment tab:
+doc-rails(dev)> Enrollment.where({ course_id: 5, final_grade: nil })
+  Enrollment Load (0.4ms)  SELECT "enrollments".* FROM "enrollments" WHERE "enrollments"."course_id" = 5 AND "enrollments"."final_grade" IS NULL /* loading for pp */ LIMIT 11 /*application='DocRails'*/
+=> 
+[#<Enrollment:0x000000010d892a98
+  id: 73,
+  course_id: 5,
+  student_id: 43,
+  final_grade: nil,
+  created_at: "2025-03-10 00:18:01.239841000 +0000",
+  updated_at: "2025-03-10 00:18:01.239841000 +0000">,
+ #<Enrollment:0x000000010d892958
+  id: 76,
+  course_id: 5,
+  student_id: 46,
+  final_grade: nil,
+  created_at: "2025-03-10 00:18:01.257310000 +0000",
+  updated_at: "2025-03-10 00:18:01.257310000 +0000">,
+ #<Enrollment:0x000000010d892818
+  id: 79,
+  course_id: 5,
+  student_id: 49,
+  final_grade: nil,
+  created_at: "2025-03-10 00:18:01.275227000 +0000",
+  updated_at: "2025-03-10 00:18:01.275227000 +0000">]
+
+Step2. find first mentor_id from mentor_enrollment_assignments where enrollment_id is 73,
+then get the mentor_id which is 27, then get the mentor_id 27 email address
+doc-rails(dev)> MentorEnrollmentAssignment.find_by(enrollment_id: 73)
+  MentorEnrollmentAssignment Load (0.1ms)  SELECT "mentor_enrollment_assignments".* FROM "mentor_enrollment_assignments" WHERE "mentor_enrollment_assignments"."enrollment_id" = 73 LIMIT 1 /*application='DocRails'*/
+=> 
+#<MentorEnrollmentAssignment:0x0000000109bbefe0
+ id: 73,
+ mentor_id: 25,
+ enrollment_id: 73,
+ created_at: "2025-03-10 00:18:01.646304000 +0000",
+ updated_at: "2025-03-10 00:18:01.646304000 +0000">
+doc-rails(dev)> Mentor.find_by(id: 25)
+  Mentor Load (0.1ms)  SELECT "mentors".* FROM "mentors" WHERE "mentors"."id" = 25 LIMIT 1 /*application='DocRails'*/
+=> 
+#<Mentor:0x000000010c914df0
+ id: 25,
+ first_name: "Rudolph",
+ last_name: "Cole",
+ email: "[FILTERED]",
+ max_concurrent_students: nil,
+ created_at: "2025-03-10 00:18:01.641819000 +0000",
+ updated_at: "2025-03-10 00:18:01.641819000 +0000">
+doc-rails(dev)> mentor = Mentor.find_by(id: 25)
+  Mentor Load (0.3ms)  SELECT "mentors".* FROM "mentors" WHERE "mentors"."id" = 25 LIMIT 1 /*application='DocRails'*/
+=> 
+#<Mentor:0x000000010d89b2d8
+...
+doc-rails(dev)> mentor.email
+=> "trenton_haley@okuneva-tremblay.test"
+doc-rails(dev)> mentor.id
+=> 25
+doc-rails(dev)> 
+
+Step3. find second mentor_id from mentor_enrollment_assignments where enrollment_id is 79, 
+then get the mentor_id which is 27, then get the mentor_id 27 email address
+doc-rails(dev)> MentorEnrollmentAssignment.find_by(id: 79)
```

Comment:
almost! you have this querying the `mentor_enrollment_assignments` table for its own `id` of `79`, when you should be looking for the `enrollment_id`.